### PR TITLE
remove `request` from `src/hosting` files (proxy middleware)

### PR DESCRIPTION
**Pull request by bkendall**
### Description

Replaces usage of the `request` library in the `hosting` directory with our new `apiv2`. `apiv2` needed to be extended to support query parameters and not follow redirects - tests have been added for that.

### Scenarios Tested

Tested against the emulator and a real function. Tested with the function taking too long and the abort logic taking effect. Tested both GET and POST requests.

### CODE REVIEW

File: src/apiv2.ts [hunk 1]
```diff
@@ -270,7 +275,9 @@ export class Client {
       body = (res.body as unknown) as ResT;
     } else {
       // This is how the linter wants the casting to T to work.
```
Comment (yuchenshi) [inline src/apiv2.ts:277]: Drop this comment

File: src/hosting/proxy.ts [hunk 2]
```diff
@@ -35,67 +41,94 @@ function makeVary(vary?: string): string {
...
-        Cookie: sessionCookie,
-      },
-      followRedirect: false,
-      timeout: 60000,
-    });
+    // req.url is just the full path (e.g. /foo?key=value; no origin).
+    const u = new URL(url + req.url);
+    const c = new Client({ urlPrefix: u.origin, auth: false });
+    const controller = new AbortController();
+    const timer: NodeJS.Timeout = setTimeout(() => controller.abort(), 60000);
```
Comment (yuchenshi) [inline src/hosting/proxy.ts:57]: nit: Consider using `Promise.race` with a Promise that resolves in 60s to make it less scary w.r.t race conditions

**Review by yuchenshi — APPROVED** [review]

File: src/hosting/proxy.ts (hunk 2 above)
Comment (yuchenshi) [reply src/hosting/proxy.ts:57]: Ah nevermind, I see how abort errors are handled before pipes now

File: src/apiv2.ts (hunk 1 above)
Comment (bkendall) [reply src/apiv2.ts:277]: Done.

**Pull request merged**